component/http/route.go

```diff
@@ -73,6 +75,15 @@ func (rb *RouteBuilder) WithAuth(auth auth.Authenticator) *RouteBuilder {
 	return rb
 }
 
+// WithRouteCache adds a cache to the corresponding route
+func (rb *RouteBuilder) WithRouteCache(routeCache *RouteCache) *RouteBuilder {
```

Comment:
As we have discussed, we should keep the API of the routes small.
With this approach the user needs to build the route cache outside of the builder and then pass it in, the suggestion was to have the route build i internally and privide only a extension method.
Ease of use is of paramount here, that why we introduced builders but having to manny builder is also not API friendly.